autorest/azure/azure.go

```diff
@@ -126,6 +131,58 @@ func ExtractRequestID(resp *http.Response) string {
 	return autorest.ExtractHeaderValue(HeaderRequestID, resp)
 }
 
+// GetAsyncOperation retrieves the long-running URL to poll from the passed response.
+func GetAsyncOperation(resp *http.Response) string {
+	return resp.Header.Get(http.CanonicalHeaderKey(HeaderAsyncOperation))
+}
+
+// ResponseIsLongRunning returns true if the passed response is for an Azure long-running operation.
+func ResponseIsLongRunning(resp *http.Response) bool {
+	return autorest.ResponseRequiresPolling(resp, http.StatusCreated) && len(GetAsyncOperation(resp)) > 0
```

Comment:
I think we ought to handle both `http.StatusCreated` (201)  and `http.StatusAccepted` (202) for long-running operations.

`Azure-AsyncOperation` header is present in both response status codes (unless the operation is completed right away sometimes with HTTP 201, then we should not poll). 

Also let's use `== ""` comparison for strings.